tools/resourcedocsgen/cmd/docs/registry.go

```diff
@@ -39,6 +40,76 @@ import (
 	concpool "github.com/sourcegraph/conc/pool"
 )
 
+// PackageMetadataProvider is an interface for providers that retrieve package metadata
+// from either the filesystem directory or the Pulumi Registry API.
+type PackageMetadataProvider interface {
+	// GetPackageMetadata returns metadata for a specific package
+	GetPackageMetadata(pkgName string) (*pkg.PackageMeta, error)
+	// ListPackageMetadata returns metadata for all packages
+	ListPackageMetadata() ([]*pkg.PackageMeta, error)
+}
+
+// FileSystemProvider implements PackageMetadataProvider using the local yaml data files
+// in the pulumi/registry repository.
+type FileSystemProvider struct {
+	registryDir string
+}
+
+// RegistryAPIProvider implements PackageMetadataProvider using the Pulumi API
+// to retrieve package metadata.
+type RegistryAPIProvider struct {
+	apiURL string
+}
+
+// PackageMetadata represents the API response structure for package metadata
+// from the Pulumi Registry API.
+// TODO: import this from the pulumi-service if possible.
```

Comment:
These will eventually be available in the `"github.com/pulumi/pulumi/sdk/v3/go/common/apitype"` package. I don't think this repo will ever be able to see in the service repo.